integrations/connext/hooks/use-approve-if-needed.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react'
+
+import axios from 'axios'
+import { TransactionRequest } from 'viem'
+import { useAccount } from 'wagmi'
+
+interface ApproveIfNeededArgs {
+  isMainnet: boolean
+  originDomain: string | undefined
+  assetAddress: string | undefined
+  amount: string | undefined
+}
+
+interface AxiosResponseData {
+  txRequest: TransactionRequest
+}
+
+interface IUseApproveIfNeeded {
+  request: TransactionRequest | undefined
+  isLoading: boolean
+}
+
+export const useApproveIfNeeded = ({ isMainnet, originDomain, assetAddress, amount }: ApproveIfNeededArgs): IUseApproveIfNeeded => {
+  const [request, setRequest] = useState<TransactionRequest>()
+  const [isLoading, setIsLoading] = useState(false)
+  const { address } = useAccount()
+
+  useEffect(() => {
```

Comment:
use `useQuery` from `tanstack-query`  rather than `useEffect` for async state management, check how other integrations use it